packages/nouns-api/src/services/ideas.ts

```diff
@@ -96,15 +74,31 @@ class IdeasService {
         },
       });
 
+      const ideasToUpdate: any = [];
+
       const ideaData = ideas
         .map((idea: any) => {
-          const votecount = calculateVotes(idea.votes);
-          const closed = getIsClosed(idea);
+          let votecount = idea.closingVoteCount;
+          let closed = idea.locked;
+          if (!closed) {
+            const { count, upvotes, downvotes } = calculateAllVotes(idea.votes);
+            votecount = count;
+            closed = getIsClosed(idea);
+
+            if (closed) {
+              //Trigger async event to update idea
+              ideasToUpdate.push({ ...idea, votecount, closed, upvotes, downvotes });
```

Comment:
Feels a bit dirty to be running these updates on the request to fetch the list but it should happen async to the incoming request and infrequently enough that it shouldn't be an issue or cause additional latency.